### PR TITLE
fix: check for webassembly support

### PR DESCRIPTION
**Pull request by pedrolamas**
Check if WebAssembly is supported/enabled before attempting to load Monaco Editor

Mitigates #821

Signed-off-by: Pedro Lamas <pedrolamas@gmail.com>

### CODE REVIEW

**Comment by pedrolamas** [comment]
While this fixes the problem for when the browser doesn't have WebAssembly support (or it is disabled), we should probably add a warning to use another browser or check settings on the current one!

File: src/util/is-web-assembly-supported.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const isWebAssemblySupported = () => {
+  try {
+    if (typeof WebAssembly === 'object' && typeof WebAssembly.instantiate === 'function') {
+      const module = new WebAssembly.Module(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00))
+
+      if (module instanceof WebAssembly.Module) {
+        return new WebAssembly.Instance(module) instanceof WebAssembly.Instance
+      }
+    }
```
Comment (matmen) [inline src/util/is-web-assembly-supported.ts:9]: Where's this sourced from? Wouldn't a simple check for `window.WebAssembly`/`typeof WebAssembly` work here?

Comment (pedrolamas) [reply src/util/is-web-assembly-supported.ts:9]: https://stackoverflow.com/a/47880734 😉

I'm happy to use the simpler solution shown on the last comment on that solution!

Comment (matmen) [reply src/util/is-web-assembly-supported.ts:9]: That answer does a nice job explaining it. Thanks, LGTM

**Review by matmen — APPROVED** [review]

**Pull request merged**